week5-objects-arrays/index.html

```diff
@@ -0,0 +1,273 @@
+  <br/>
+
+  1. Write a program that implements the above sketch. First, draw the first word from the <code>words</code> array on the canvas. And everytime when the mouse is pressed, draw the next word from the array <code>words</code>. We use a variable called <code>index</code> to represent the index of each word in the array. And at last, if the index equals to the length of the array, make index 0 again.
+</p>
+
+<script type="text/p5" data-preview-width="400">
+var words = ["ಥ_ಥ", "(ง'̀-'́)ง", "(°ロ°)☝", "(｡◕‿‿◕｡)", "( ͡° ͜ʖ ͡°)", "(ᵔᴥᵔ)", "༼ つ ◕_◕ ༽つ"];
+
+var index = ____;
+
+function setup() {
+  createCanvas(200, 200);
+}
+
+function draw() {
+  background(0);
+
+  fill(255);
+  textSize(32);
+  text(words[____], 10, 100);
+}
+
+function mousePressed() {
+  ____ = ____ + ____;
+
+  if (____ == ________) {
+    ____ = ____;
+  }
+}
+</script>
+
+<p>
+  <iframe src="https://alpha.editor.p5js.org/embed/Syfs_VqC" width=300 height=200></iframe>
+  <br/>
+
+  2. Assuming an array of 10 integers, that is, <code>var nums = [ 5, 4, 2, 7, 6, 8, 5, 2, 8, 14 ];</code> Write code to perform the following array operations. The sketch above shows the final results.<br/><br/>
+
+  (a) Square each number (i.e., multiply each by itself).<br/>
+  (b) Add a random number between zero and 10 to each number.<br/>
+  (c) Add to each number the number that follows in the array. Skip the last value in the array.<br/>
+  (d) Calculate the sum of all the numbers.<br/><br/>
+
+  (Note that the number of clues vary, just because a [____] is not explicitly written in does not mean there should not be brackets).</p>
+
+<script type="text/p5" data-preview-width="400">
+var nums = [ 5, 4, 2, 7, 6, 8, 5, 2, 8, 14 ];
+
+function setup() {
+  createCanvas(300, 200);
+  background(255);
+  
+  // Square each number (i.e., multiply each by itself )
+  for (var i ___; i < ____; i ++ ) {
+    ___[i] = _____*_____;
+    text(nums[i]+',', i * 25 + 10, 50);
+  }
+  
+  // Add a random number between zero and 10 to each number.
+  ________________________________
+    ____ += int(________);
+    text(nums[i]+',', i * 25 + 10, 80);
+  }
+
+  // Add to each number the number that follows in the array. Skip the last value in the array. 
+  for (var i = 0; i < ____; i ++ ) {
+    _____ += ______ [____];
+    text(nums[i]+',', i * 25 + 10, 110);
+  }
+
+  // Calculate the sum of all the numbers.
+  ____ ________ = ___;
+  for (var i = 0; i < nums.length; i ++ ) {
+    ______ += ________;
+  }
+  text('sum: ' + sum, 10, 140);
+}
+</script>
+
+
+<p>
+  <iframe src="https://alpha.editor.p5js.org/embed/HkrRhEcR" width=200 height=200></iframe>
+  <br/>
+
+  3. Write a program that implements the above sketch. Use an array called <code>bubbles</code> and a for loop to store and get access to each bubble object.</p>
+
+<script type="text/p5" data-preview-width="400">
+var bubbles = [];
+
+function setup() {
+  createCanvas(200, 200);
+  for (___________________) {
+    bubbles[____] = {
```

Comment:
we're not using this syntax anymore in ICM.  We are focusing on the constructor function only. So I would change this to maybe step 1 making a single object with a constructor function and then have the next one be the array of objects made with the constructor function.